cfsdk/src/main/java/io/harness/cfsdk/CfClient.java

```diff
@@ -873,43 +379,16 @@ public JSONObject jsonVariation(String evaluationId, JSONObject defaultValue) {
     }
 
 
-    /**
-     * Adds new listener for various SDK events. See {@link StatusEvent.EVENT_TYPE} for possible types.
-     *
-     * @param observer {@link EventsListener} implementation that will be triggered when there is a change in state of SDK
-     * @return Was listener registered with success?
-     */
-    public boolean registerEventsListener(final EventsListener observer) {
-
-        if (observer != null) {
-
-            return eventsListenerSet.add(observer);
-        }
-        return false;
-    }
-
-    /**
-     * Removes registered listener from list of registered events listener
-     *
-     * @param observer {@link EventsListener} implementation that needs to be removed
-     * @return Was listener un-registered with success?
-     */
-    public boolean unregisterEventsListener(final EventsListener observer) {
-
-        return eventsListenerSet.remove(observer);
-    }
-
     /**
      * Clears the occupied resources and shut's down the sdk.
      * After calling this method, the {@link #initialize} must be called again. It will also
      * remove any registered event listeners.
      */
     @Override
-    public void destroy() {
+    public void close() {
```

Comment:
Definitely a better name. Forgive me if this is a stupid question, but if we change the public method name here, will that not break backwards compatability?

Reply:
Oh nevermind. See the deprecated for `destroy` below! Ignore.

Reply:
yeah the main problem here is destroy() is non-standard and can't be used in try-with-resources statements. it was re-added it back in as deprecated to allow developers time to upgrade. then we can remove it later